llvm/lib/IR/Instruction.cpp

```diff
@@ -318,10 +318,7 @@ void Instruction::setIsExact(bool b) {
 }
 
 void Instruction::setIsDisjoint(bool b) {
-  assert(isa<PossiblyDisjointInst>(this) && "Must be or");
-  SubclassOptionalData =
-      (SubclassOptionalData & ~PossiblyDisjointInst::IsDisjoint) |
-      (b * PossiblyDisjointInst::IsDisjoint);
+  cast<PossiblyDisjointInst>(this)->setIsDisjoint(b);
```

Comment:
I believe the suggestion was to not have these methods on Instruction, only the ones on PossiblyDisjointInst.